test/e2e/framework/job/wait.go

```diff
@@ -68,6 +68,22 @@ func WaitForJobComplete(ctx context.Context, c clientset.Interface, ns, jobName
 	})
 }
 
+// WaitForJobSuspend uses c to wait for suspend condition for the Job jobName in namespace ns.
+func WaitForJobSuspend(ctx context.Context, c clientset.Interface, ns, jobName string) error {
+	return wait.PollUntilContextCancel(ctx, framework.Poll, true, func(ctx context.Context) (bool, error) {
```

Comment:
@kannon92 WaitFor + something  isn't a good pattern for new test cases, see https://github.com/kubernetes/kubernetes/issues/115164 for details. please consider using gomega.Eventually to handle this case, https://github.com/kubernetes/kubernetes/pull/119695 is a good example.

Reply:
Hmm.  I see.  I did not know that this was considered an antipattern.

We can close this PR as is if this is the better approach for testing.

Reply:
/cc @pohly 
can you take a look at it?

Reply:
In this test we need some form of waiting as the Job controller needs time to add the condition, based on the  `.spec.suspend` field, which happens here: https://github.com/kubernetes/kubernetes/blob/3d1ac86013db78a270a5dedbdedc63027333aef4/pkg/controller/job/job_controller.go#L893-L900, and is later send to the api server.

Currently, this waiting is hidden inside `Consistently` which verifies the pods aren't created. IMO, it improves readability to have some explicit wait for the suspend field. So, I would suggest just adjust the PR to use `gomega.Eventually`. IIUC using `WaitForJobSuspend` or  `gomega.Eventually` is more of a technical detail.

Reply:
> I would suggest just adjust the PR to use gomega.Eventually.

Agreed.